bindings/rust/src/lib.rs

```diff
@@ -8,11 +8,35 @@
 
 mod sys;
 
+use std::ptr::NonNull;
+
 /// Parse and validate the input according to WebAssembly 1.0 rules. Returns true if the supplied input is valid.
 pub fn validate<T: AsRef<[u8]>>(input: T) -> bool {
     unsafe { sys::fizzy_validate(input.as_ref().as_ptr(), input.as_ref().len()) }
 }
 
+/// A parsed and validated WebAssembly 1.0 module.
+pub struct Module {
+    ptr: NonNull<sys::FizzyModule>,
+}
+
+impl Drop for Module {
+    fn drop(&mut self) {
+        unsafe { sys::fizzy_free_module(self.ptr.as_ptr()) }
+    }
+}
+
+/// Parse and validate the input according to WebAssembly 1.0 rules.
+pub fn parse<T: AsRef<[u8]>>(input: &T) -> Result<Module, ()> {
+    let ptr = unsafe { sys::fizzy_parse(input.as_ref().as_ptr(), input.as_ref().len()) };
+    if ptr.is_null() {
```

Comment:
Can you use `NonNull::new()` here instead, and then check the `Option`. Should be on `unsafe` less.

Reply:
`Module` is not using the `NonNull` wrapping.

Reply:
I'm trying to remember. I think it can use `NonNull` again since I'm using `forget()`.

Reply:
Ok, the reason I'm using `NonNull` for `Module` is because [NonNull requires *mut](https://doc.rust-lang.org/std/ptr/struct.NonNull.html#method.new), while Module is *const.

Reply:
The first commit had it with `NonNull` and doing a typecast..

Reply:
I added this comment to the first commit, later outdated. So this is resolved.